### PR TITLE
Fixed #24372 - Replaced TokenParser usage with traditional parsing.

### PR DESCRIPTION
**Pull request by prestontimmons**

### CODE REVIEW

File: django/templatetags/i18n.py [hunk 1]
```diff
@@ -348,42 +345,57 @@ def do_translate(parser, token):
...
+            raise TemplateSyntaxError(
+                "The '%s' option was specified more than once." % option,
+            )
+        elif option == 'noop':
+            noop = True
+        elif option == 'context':
+            try:
+                value = remaining.pop(0)
+            except IndexError:
+                msg = ("No argument provided to the '%s' tag for the context option.") % bits[0]
```
Comment (timgraham) [inline django/templatetags/i18n.py]: no parens needed here and a couple lines below

File: django/templatetags/i18n.py [hunk 2]
```diff
@@ -348,42 +345,57 @@ def do_translate(parser, token):
...
+            message_context = parser.compile_filter(value)
+        elif option == 'as':
+            try:
+                value = remaining.pop(0)
+            except IndexError:
+                msg = ("No argument provided to the '%s' tag for the as option.") % bits[0]
+                six.reraise(TemplateSyntaxError, TemplateSyntaxError(msg), sys.exc_info()[2])
+            asvar = value
+        else:
+            msg = (
```
Comment (timgraham) [inline django/templatetags/i18n.py]: remove intermediate `msg` variable for consistency with other `TemplateSyntaxError`s?

File: django/templatetags/i18n.py [hunk 3]
```diff
@@ -348,42 +345,57 @@ def do_translate(parser, token):
...
+        raise TemplateSyntaxError("'%s' takes at least one argument" % bits[0])
+    message_string = parser.compile_filter(bits[1])
+
+    remaining = bits[2:]
+
+    noop = False
+    asvar = None
+    message_context = None
+    seen = set()
+
```
Comment (timgraham) [inline django/templatetags/i18n.py]: is this blank line meaningful? (is `invalid_context` different from other vars above?)

**Comment by prestontimmons** [comment]
Thanks. I updated the formatting.

**Comment by timgraham** [comment]
merged in 358850781f8fd3d79aba5b1a9a0b8d28f544bf8a, thanks!

**Pull request closed without merging**